packages/yoga/package.json

```diff
@@ -5,9 +5,10 @@
   "main": "./src",
   "sideEffects": false,
   "scripts": {
-    "build": "yarn build:web && yarn build:native && yarn build:esm",
-    "build:web": "NODE_ENV=web babel ./src --out-dir dist/cjs",
-    "build:esm": "NODE_ENV=esm babel ./src --out-dir dist/esm",
+    "build": "yarn build:types && yarn build:cjs && yarn build:esm",
+    "build:types": "tsup --outDir dist/typings --dts-only",
+    "build:cjs": "tsup --outDir dist/cjs --format=cjs",
+    "build:esm": "tsup --format=esm --legacy-output",
```

Comment:
Can we have this in a separate PR?
@lsantos1873 is working on a new component (SideMenu) and we would like it to be done already in TS. 

Is that possible? :)

Reply:
Yo @MicaelRodrigues , how are you? 😁

Yes, that would be nice! Thanks for the suggestion, I'll work on it.

This PR ended up being pretty big, so I'm going to split it into other PRs. 🛠️